service/src/test/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceTest.java

```diff
@@ -184,6 +180,45 @@ public void testEvaluateWithPolicyAndSubjectResourceAttributes(final String acsS
         }
     }
 
+    @Test(
+            dataProvider = "filterPolicySetsInvalidRequestDataProvider",
+            expectedExceptions = IllegalArgumentException.class)
+    public void testFilterPolicySetsByPriorityForInvalidRequest(final List<PolicySet> allPolicySets,
+            final List<String> policySetsPriority) throws JsonParseException, JsonMappingException, IOException {
+        this.evaluationService.filterPolicySetsByPriority("subject1", "resource1", allPolicySets, policySetsPriority);
+    }
+
+    @Test(dataProvider = "filterPolicySetsDataProvider")
+    public void testFilterPolicySetsByPriority(final List<PolicySet> allPolicySets,
+            final List<String> policySetsPriority, List<PolicySet> expectedFilteredPolicySets)
+            throws JsonParseException, JsonMappingException, IOException {
+        List<PolicySet> actualFilteredPolicySets = this.evaluationService.filterPolicySetsByPriority("subject1",
+                "resource1", allPolicySets, policySetsPriority);
+        Assert.assertEquals(actualFilteredPolicySets, expectedFilteredPolicySets);
+    }
+
+    @Test(dataProvider = "multiplePolicySetsRequestDataProvider")
+    public void testEvaluateWithMultiplePolicySets(final List<PolicySet> allPolicySets,
+            final List<String> policySetsPriority, final Effect effect)
+                    throws JsonParseException, JsonMappingException, IOException {
+        when(this.policyService.getAllPolicySets()).thenReturn(allPolicySets);
+        when(this.policyMatcher.matchForResult(any(PolicyMatchCandidate.class), anyListOf(Policy.class)))
+                .thenAnswer(new Answer<MatchResult>() {
+                    public MatchResult answer(final InvocationOnMock invocation) {
+                        Object[] args = invocation.getArguments();
+                        @SuppressWarnings("unchecked")
+                        List<Policy> policyList = (List<Policy>) args[1];
+                        List<MatchedPolicy> matchedPolicies = new ArrayList<>();
+                        policyList.forEach(policy -> matchedPolicies.add(new MatchedPolicy(policy, EMPTY_ATTRS, EMPTY_ATTRS)));
```

Comment:
If the intent is to match all policies, this code can be removed by using policies which have no targets. It will make the test easier to understand.